### PR TITLE
Consider SIP content type to determine if the next layer is SDP (fix #1309)

### LINKED ISSUE #1309: Sip content type not only can be sdp
In file `Packet++/src/SipLayer.cpp:84`:
the function `SipLayer::parseNextLayer()` while inaccurate parse next layer, when `Content-Type` is not `application/sdp`.
The value of `Content-Type` may be `multipart/mixed` or `message/sip` etc.

### PR DESCRIPTION
**Pull request by Double0101**
In issue https://github.com/seladb/PcapPlusPlus/issues/1309
Cause:
When sip content-type is message/sip, the function SipLayer::parseNextLayer() will parse next layer as sdp inappropriate.
Solution:
Check sip content-type before parse next layer as sdp.

### CODE REVIEW

**Comment by tigercosmos** [comment]
thanks for contributing, could you take a look at why the CI fails?

File: Packet++/src/SipLayer.cpp [hunk 1]
```diff
@@ -87,8 +87,15 @@ void SipLayer::parseNextLayer()
...
 	size_t headerLen = getHeaderLen();
+	std::string contentType;
 	if (getContentLength() > 0)
 	{
+		HeaderField *contentTypeField = getFieldByName(PCPP_SIP_CONTENT_TYPE_FIELD);
+		if (contentTypeField != nullptr)
+			contentType = contentTypeField->getFieldValue();
+	}
+
+	if (contentType.find("application/sdp") != std::string::npos) {
```
Comment (tigercosmos) [inline Packet++/src/SipLayer.cpp:98]: nit: we use 
```
if
{
...
}
```

**Comment by tigercosmos** [comment]
please also add tests to check if "Content-Type" is not sdp.

File: Packet++/src/SipLayer.cpp
Comment (seladb) [inline Packet++/src/SipLayer.cpp:1]: @Double0101 can we add a test for it?

Comment (Double0101) [reply Packet++/src/SipLayer.cpp:1]: I'm not familiar with `.dat` files. So I'm trying to create a suitable test packet example.

Comment (tigercosmos) [reply Packet++/src/SipLayer.cpp:1]: it's just a hex format of the packet bytes.

so, if you use any hex to ASCII tool, you will see the text content of the packet from `.dat` file.

if you want to generate a `.dat` file, a possible way is that you can use Wireshark and dump the packet bytes as hex.

File: Tests/Packet++Test/PacketExamples/sip_not_sdp.dat [hunk 2]
```diff
@@ -0,0 +1 @@
+00908f0a7ccc00908f0a393a080045a003a10c8700004011491b0a2106640a21066513c413c4038db3595349502f322e3020323030204f4b0d0a5669613a205349502f322e302f5544502031302e33332e362e3130313b6272616e63683d7a39684734624b6163313435303533333939310d0a46726f6d3a203c7369703a3230314031302e33332e362e3130313e3b7461673d3163313435303533303934330d0a546f3a203c7369703a3130314031302e33332e362e3130303b757365723d70686f6e653e3b7461673d31633738323630393332310d0a43616c6c2d49443a20313435303533303337373135323230313036323232314031302e33332e362e3130310d0a435365713a203120494e564954450d0a436f6e746576e742d547970653a206170706c69636174696f6e2f766e642e336770702e736d730d0a436f6e74656e742d4c656e6774683a2032370d0a0d0a0002000891683110700205f00e016c05810110f00000046c367e0c0102030311d1c127955d7f0ff5db98a9c1
```
Comment (tigercosmos) [inline Tests/Packet++Test/PacketExamples/sip_not_sdp.dat:1]: it would be nice if you can provide the text of this hex.

Comment (Double0101) [reply Tests/Packet++Test/PacketExamples/sip_not_sdp.dat]: Ok, I upload a pcap file in `PacketExamples` directory.

File: Tests/Packet++Test/Tests/SipSdpTests.cpp [hunk 3]
```diff
@@ -648,6 +648,22 @@ PTF_TEST_CASE(SdpLayerParsingTest)
 } // SdpLayerParsingTest
 
 
+PTF_TEST_CASE(NotSdpLayerParseingTest)
+{
+	timeval time;
```
Comment (tigercosmos) [inline Tests/Packet++Test/Tests/SipSdpTests.cpp:653]: we don't need this?

Comment (tigercosmos) [reply Tests/Packet++Test/Tests/SipSdpTests.cpp:653]: please remove unused code.

Comment (Double0101) [reply Tests/Packet++Test/Tests/SipSdpTests.cpp:653]: I see that the macro `READ_FILE_AND_CREATE_PACKET` need the argument `time`, so this argument needs to be retained.

Comment (tigercosmos) [reply Tests/Packet++Test/Tests/SipSdpTests.cpp:653]: yeah... it's a weird design ...

**Review by tigercosmos — APPROVED** [review]

File: Tests/Packet++Test/Tests/SipSdpTests.cpp [hunk 4]
```diff
@@ -648,7 +648,7 @@ PTF_TEST_CASE(SdpLayerParsingTest)
 } // SdpLayerParsingTest
 
 
-PTF_TEST_CASE(NotSdpLayerParseingTest)
+PTF_TEST_CASE(NotSdpLayerParsingTest)
```
Comment (tigercosmos) [inline Tests/Packet++Test/Tests/SipSdpTests.cpp:651]: btw, you need to update here to run the test:
- Tests/Packet++Test/main.cpp
- TestDefinition.h

Comment (Double0101) [reply Tests/Packet++Test/Tests/SipSdpTests.cpp:651]: Ok, I nailed it.

File: Tests/Packet++Test/Tests/SipSdpTests.cpp [hunk 5]
```diff
@@ -648,6 +648,22 @@ PTF_TEST_CASE(SdpLayerParsingTest)
...
+	timeval time;
+	gettimeofday(&time, nullptr);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/sip_not_sdp.dat");
+
+	pcpp::Packet notSdpPacket(&rawPacket1);
+
+	PTF_ASSERT_TRUE(notSdpPacket.isPacketOfType(pcpp::SIP));
+	pcpp::SipRequestLayer* sipLayer = notSdpPacket.getLayerOfType<pcpp::SipRequestLayer>();
+	PTF_ASSERT_EQUAL(sipLayer->getContentLength(), 273);
```
Comment (seladb) [inline Tests/Packet++Test/Tests/SipSdpTests.cpp:662]: We can also check that the layer after `SipRequestLayer` is of type `PayloadLayer`

File: Tests/Packet++Test/TestDefinition.h [hunk 6]
```diff
@@ -150,6 +150,7 @@ PTF_TEST_CASE(SipResponseLayerParsingTest);
 PTF_TEST_CASE(SipResponseLayerCreationTest);
 PTF_TEST_CASE(SipResponseLayerEditTest);
 PTF_TEST_CASE(SdpLayerParsingTest);
+PTF_TEST_CASE(NotSdpLayerParsingTest);
```
Comment (seladb) [inline Tests/Packet++Test/TestDefinition.h:153]: I suggest renaming the test to `SipNotSdpLayerParsingTest`

**Comment by tigercosmos** [comment]
@Double0101  you don't need to force push, we will squash commits anyway. better to keep the commit to easily track your changes.

**Comment by tigercosmos** [comment]
LGTM

**Review by seladb — APPROVED** [review]

**Pull request merged**